src/business/form.ts

```diff
@@ -1,44 +1,71 @@
+export interface FieldState {
+    value: any;
+    validationError?: string;
+    pristine: boolean;
+}
+
 export class Form {
 
-    private values: { [key: string]: any } = {};
+    private fieldStates: { [key: string]: FieldState } = {};
+    private fieldListeners: { [fieldName: string]: Array<(value: any) => void> } = {};
 
-    private validationErrors: { [fieldName: string]: any } = {};
+    public getFieldState(fieldName: string): FieldState | undefined {
+        return this.fieldStates[fieldName];
+    }
 
-    private fieldListeners: { [fieldName: string]: Array<(value: any) => void> } = {};
+    public initialiseFieldValues(values: { [fieldName: string]: any }) {
+        const fieldNames = Object.keys(values);
+        fieldNames.forEach((fieldName) => {
+            this.updateFieldValue(fieldName, values[fieldName], true);
+        });
 
-    public getFieldValue(fieldName: string): any {
-        return this.values[fieldName] || '';
+        this.triggerMultipleFieldListeners(fieldNames);
     }
 
     public setFieldValues(values: { [fieldName: string]: any }) {
         const fieldNames = Object.keys(values);
         fieldNames.forEach((fieldName) => {
-            this.values[fieldName] = values[fieldName];
-            delete this.validationErrors[fieldName];
+            this.updateFieldValue(fieldName, values[fieldName], false);
         });
 
         this.triggerMultipleFieldListeners(fieldNames);
     }
 
+    public initialiseFieldValue(fieldName: string, value: any) {
+        this.updateFieldValue(fieldName, value, true);
+
+        this.triggerFieldListeners(fieldName);
+    }
+
     public setFieldValue(fieldName: string, value: any) {
-        this.values[fieldName] = value;
-        delete this.validationErrors[fieldName];
+        this.updateFieldValue(fieldName, value, false);
 
         this.triggerFieldListeners(fieldName);
     }
 
+    private updateFieldValue(fieldName: string, value: any, pristineValue: boolean) {
```

Comment:
I'm expecting the value to be named `pristineValue`, do you mean `isPristineValue`?